EIPS/eip-5299.md

```diff
@@ -0,0 +1,173 @@
+---
+eip: 5299
+title: Layer3 scaling pattern
+description: A primitive to facilitate eternal upgradeability of smart contracts, by making storage slots dynamic.
+author: Vinay Kumar (@v9hstk), Pratik Agarwal (@Prtk418)
+discussions-to: https://ethereum-magicians.org/t/eip-9000-layer3-scaling-standard/9039
+status: Draft
+type: Standards Track
+category: ERC
+created: 2021-07-18
+---
+
+## Abstract
+Upgrading a storage layout is required in case of a complex project while it’s re architected. The EIP introduces implementation of dynamic storage slots, which can be upgraded to a new one in case of an upgrade. Without actually deploying a new contract altogether.
```

Comment:
```suggestion
Upgrading a storage layout is required in case of a complex project if it is re-architected. The EIP introduces dynamic storage slots, which can be upgraded without deploying a new contract.
```